src/subscriptions/SubscriptionsContainer.js

```diff
@@ -5,12 +5,25 @@ import type { GlobalState } from '../types';
 import { getAuth } from '../selectors';
 import boundActions from '../boundActions';
 import SubscriptionsCard from './SubscriptionsCard';
+import { getUnreadStreams } from '../baseSelectors';
+import { queueMarkAsRead } from '../api';
 
 export default connect(
   (state: GlobalState) => ({
     auth: getAuth(state),
     streams: state.streams,
     subscriptions: state.subscriptions,
+    markStreamMessagesRead: (streamId: number) => {
+      const removeIds = [].concat(
+        ...getUnreadStreams(state)
+          .filter(stream => streamId === stream.stream_id)
+          .map(stream => stream.unread_message_ids),
+      );
```

Comment:
Optionally, but we tend to do it now, we do a check of `length` of input vs output after the `filter`. If the lengths didn't change, we do not need to use the new value as it did not change.